src/types/iterator.rs

```diff
@@ -213,4 +229,30 @@ mod tests {
         let iter: &PyIterator = PyIterator::try_from(obj.as_ref(py)).unwrap();
         assert_eq!(obj, iter.into());
     }
+
+    #[test]
+    fn test_as_ref() {
+        let gil = Python::acquire_gil();
+        let py = gil.python();
+        let iter: Py<PyIterator> = PyList::empty(py).to_object(py).as_ref(py).iter().unwrap().into();
```

Comment:
I think this can skip the `to_object(py)` part like this:

```suggestion
        let iter: Py<PyIterator> = PyAny::iter(PyList::empty(py)).unwrap().into();
```